modules/samples/artifacts/CSVMappingWithInmemoryTransport/CSVMappingWithInmemoryTransport.siddhi

```diff
@@ -0,0 +1,90 @@
+
+Prerequisites:
+    1) Save this sample
+
+Executing the Sample:
+    1) Start the Siddhi application by clicking on 'Run'.
+    2) If the Siddhi application starts successfully, the following messages would be shown on the console.
+        * Tcp Server started in 0.0.0.0:9892
+        * CSVMappingWithInmemoryTransport.siddhi - Started Successfully!
+
+Testing the Sample:
+    Send events through one or more of the following methods.
+
+    Option1: Publish events with tcp sample client:
+    Run "ant" command in the terminal as follows:
+        ant -Dtype=binary
+    If you want to publish custom number of events, you need to run "ant" command as follows
+        ant -Dtype=binary -DnoOfEventsToSend=2
+
+    Option2:Send events with tcp server through the event simulator:  
+       * Send events with http server through the event simulator:
+        a) Open the event simulator by clicking on the second icon or pressing Ctrl+Shift+I.
+    	b) In the Single Simulation tab of the panel, specify the values as follows:
+                * Siddhi App Name  : CSVMappingWithInmemoryTransport
+                * Stream Name     : SweetProductionStream
+        c) In the 'id' and 'amount' fields, enter 'toffees' and '123.5' respectively and then click Send to send the event.
+        d) Send more events as desired.
+
+Viewing the Results:
+    INFO {org.wso2.siddhi.core.stream.output.sink.LogSink} - Default Mapper : Event{timestamp=1517987458003, data=[Jelly, 123.3], isExpired=false} 
+    INFO {org.wso2.siddhi.core.stream.output.sink.LogSink} - Custom Mapper : Event{timestamp=1517987458003, data=[Jelly, 123.3], isExpired=false}
+    INFO {org.wso2.siddhi.core.stream.output.sink.LogSink} - Default Mapper : Event{timestamp=1517987458003, data=[Cupcake, 20.5], isExpired=false} 
+    INFO {org.wso2.siddhi.core.stream.output.sink.LogSink} - Custom Mapper : Event{timestamp=1517987458003, data=[Cupcake, 20.5], isExpired=false} 
+
+Notes:
+    If the message "Tcp Server started in 0.0.0.0:9892" does not appear,it could be due to port 9892, defined in the Siddhi application is already being used by a different program. To resolve this issue, please do the following,
+        * Stop this Siddhi application (Click 'Run' on menu bar -> 'Stop')
+        * Change the port 9892 to an unused port, in this Siddhi application's source configuration.
+        * Start the application and check whether the specified messages appear on the console
+*/
+
+@source(type='tcp', context='SweetProductionStream', port='9892',
+@map(type='binary'))
+define stream SweetProductionStream (name string, amount double);
+
+@sink(type='log', prefix='Default Mapper')
+define stream DefaultOutputStream (name string, amount double);
+
+@sink(type='log', prefix='Custom Mapper')
+define stream CustomOutputStream (name string, amount double);
+
+-- Default CSV mapping.
+
+@sink(type='inMemory', topic='home', @map(type='csv'))
```

Comment:
Shall we make the source and sink as file

Reply:
its better to have two samples one for custom mapping and one for default mapping.